lib/util.js

```diff
@@ -1057,3 +1057,68 @@ exports._exceptionWithHostPort = function(err,
 // process.versions needs a custom function as some values are lazy-evaluated.
 process.versions[exports.inspect.custom] =
   (depth) => exports.format(JSON.parse(JSON.stringify(process.versions)));
+
+const kCustomCallbackifyiedSymbol = Symbol('util.callbackify.custom');
+const kCustomCallbackifyiedToSplat = Symbol('util.callbackify.toSplat');
+const kCustomCallbackifyiedProject = Symbol('util.callbackify.project');
+
+function callbackify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to callbackify() ' +
+      'must be a function');
+  }
+
+  if (orig[kCustomCallbackifyiedSymbol]) {
+    const fn = orig[kCustomCallbackifyiedSymbol];
+    if (typeof fn !== 'function') {
+      throw new TypeError('The [util.callbackify.custom] property must be ' +
+        'a function');
+    }
+    Object.defineProperty(fn, kCustomCallbackifyiedSymbol, {
```

Comment:
I'd protect this with `if (fn[kCustomCallbackifyiedSymbol] !== fn)` as `Object.defineProperty` may be quite slow.